src/app/attendance/components/AttendanceFormBuilder.tsx

```diff
@@ -162,112 +165,123 @@ const AttendanceFormBuilder = () => {
     form.setValue("memberId", Number(memberId), { shouldValidate: true });
   };
 
-  const onSubmit = (values: AttendanceFormValues) => {
-    console.log({ values });
+  const onSubmit = async (values: AttendanceFormValues) => {
+    try {
+      const result = await onAttandanceRequset(values);
+      if (!result) {
+        form.reset();
+        toast({
+          title: "Attendance submit successfully",
+          description: "You will receive message shortly",
+        });
+      }
+    } catch (error) {
+      console.error("Submission error:", error);
+      toast({
+        title: "Something went wrong",
+      });
+      return;
+    }
   };
 
   return (
-    <Form {...form}>
-      <form
-        onSubmit={form.handleSubmit(onSubmit)}
-        className="max-w-lg mx-auto p-8 mt-10 bg-white shadow-md rounded-md space-y-6"
-      >
-        <h1 className="text-2xl font-bold text-center mb-6">Attendance Form</h1>
+    <FormWrapper>
+      <Form {...form}>
+        <form
+          action={() => form.handleSubmit(onSubmit)()}
+          className="max-w-lg mx-auto p-8 mt-10 bg-white shadow-md rounded-md space-y-6"
+        >
+          <h1 className="text-2xl font-bold text-center mb-6">
+            Attendance Form
+          </h1>
 
-        <FormField
-          control={form.control}
-          name="memberId"
-          render={({ field }) => (
-            <FormItem>
-              <FormLabel>User Name</FormLabel>
-              <FormControl>
-                <Select
-                  value={field.value?.toString() || ""}
-                  onValueChange={handleUserSelect}
-                  disabled={loginUser.role === "member"}
-                >
-                  <SelectTrigger
-                    className={`border rounded-md p-2 ${
-                      form.formState.errors.memberId
-                        ? "border-red-500"
-                        : "border-gray-300"
-                    }`}
+          <FormField
+            control={form.control}
+            name="memberId"
+            render={({ field }) => (
+              <FormItem>
+                <FormLabel>User Name</FormLabel>
+                <FormControl>
+                  <Select
+                    value={field.value?.toString() || ""}
```

Comment:
This doesn't look correct. Why need to transform into string? Isn't it already string?

Reply:
No, the default value is a number; we submit the member ID, not the name. However, in the UI, we display the member name in the select

Reply:
We should do the mapping before looping.